cpp/src/io/parquet/page_delta_decode.cu

```diff
@@ -579,17 +580,27 @@ CUDF_KERNEL void __launch_bounds__(decode_block_size)
     __syncthreads();
   }
 
-  // Now turn the array of lengths into offsets, but skip if this is a large string column. In the
-  // latter case, offsets will be computed during string column creation.
-  if (not s->col.is_large_string_col) {
-    int value_count = nesting_info_base[leaf_level_index].value_count;
+  // Number of values decoded from this page
+  int value_count = nesting_info_base[leaf_level_index].value_count;
 
-    // if no repetition we haven't calculated start/end bounds and instead just skipped
-    // values until we reach first_row. account for that here.
-    if (!has_repetition) { value_count -= s->first_row; }
+  // if no repetition we haven't calculated start/end bounds and instead just skipped
+  // values until we reach first_row. account for that here.
+  if (!has_repetition) { value_count -= s->first_row; }
```

Comment:
nit:
```cuda
  auto const value_count = [&](){
    int res = nesting_info_base[leaf_level_index].value_count;
    if (!has_repetition) { res -= s->first_row; }
    return res;
  }();
```

Also, it seems this logic is repeated several times in different places. Could we make this a common utility somewhere?

Reply:
All done!